### PR TITLE
Switch to JarInJar

### PR DESCRIPTION
**Pull request by tmvkrpxl0**
This PR migrates KFF from Shadowing to JarInJar
Also this fixes KFF not being launched

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -61,47 +43,81 @@ for (s in arrayOf(sourceSets.main, sourceSets.test)) {
...
-    library("org.jetbrains.kotlinx:kotlinx-coroutines-core-jvm:$coroutines_version")
-    library("org.jetbrains.kotlinx:kotlinx-coroutines-jdk8:$coroutines_version")
-    library("org.jetbrains.kotlinx:kotlinx-serialization-json:$serialization_version")
+    minecraft("net.minecraftforge:forge:1.19-41.0.91")
+
+    val library = configurations["library"]
+
+    library("org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlin_version") {
+        exclude(group = "org.jetbrains", module = "annotations")
+        jarJar(group = group!!, name = name, version = "[$version, 2.0)") {
```
Comment (thedarkcolour) [inline build.gradle.kts]: Why is the version negotiation up to 2.0 only?

File: src/main/kotlin/thedarkcolour/kotlinforforge/forge/KDeferredRegister.kt [hunk 2]
```diff
@@ -30,18 +29,12 @@ public typealias KDeferredRegister<T> = DeferredRegister<T>
...
-public inline fun <V, T : V> KDeferredRegister<V>.registerObject(
+public fun <V, T : V> KDeferredRegister<V>.registerObject(
     name: String,
-    noinline supplier: () -> T
-): ReadOnlyProperty<Any?, T> {
+    supplier: () -> T
+): ObjectHolderDelegate<T> {
     val registryObject = this.register(name, supplier)
 
     // note that this anonymous class inherits three types
```
Comment (thedarkcolour) [inline src/main/kotlin/thedarkcolour/kotlinforforge/forge/KDeferredRegister.kt]: This should comment be removed too now that there is a concrete type

**Review by thedarkcolour — COMMENTED** [review]
Do all the inline functions now work without inline? In game too?

**Comment by tmvkrpxl0** [comment]
No, I thought they worked but they don't, they have to be game library.
I'll try to separate kff into 2 modules as I stated before, and send another PR then
Some of changes of this PR must be reverted back to inline ones, but my laptop battery died, I'll revert those soon

**Comment by thedarkcolour** [comment]
I think replacing Shadow with JiJ would be good as one pull request and the GAMELIBRARY module can be moved into a separate PR

File: src/main/kotlin/thedarkcolour/kotlinforforge/forge/KDeferredRegister.kt (hunk 2 above)
Comment (tmvkrpxl0) [reply src/main/kotlin/thedarkcolour/kotlinforforge/forge/KDeferredRegister.kt]: KDeferredRegister.kt is now reverted

File: build.gradle.kts (hunk 1 above)
Comment (tmvkrpxl0) [reply build.gradle.kts]: Now different libs use different versions

**Pull request merged**